### ISSUE TITLE
Fail to download delta image with ENOMEM

### ISSUE THREAD

**Issue by imrehg** [resin-io/resin-supervisor#495]
During resininHUP 1.x->2.x the user application is removed because of required file system changes. After the update, the device reboots with no docker images. If deltas are enabled, the supervisor (after it was downloaded itself) will try to download the delta, compared to nothing. 

Had a particular device, that couldn't finish that, as the download progress bar went all the way to the top, then it failed with:

```
Failed to download application 'registry2.resin.io/XXXX/abcdef' due to 'spawn ENOMEM'
```
This repeated one after the other. The supervisor is `v6.2.5` and resinOS 2.3.0 on Raspberry Pi 3.

Just before the error, at 99% download, the memory was like this, based on `top`:
```
Mem: 935356K used, 11476K free, 5992K shrd, 48672K buff, 277824K cached
```

<img src="https://frontapp.com/assets/img/icons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_2bjv)

**Comment by dfunckt** [comment]
fixed by https://github.com/resin-io-modules/resumable-request/pull/4

will be included in Supervisor 6.3.3 or 6.3.4

**Comment by imrehg** [comment]
@dfunckt thanks a lot! I guess this will limit which versions it's possible to update to using 1x->2x updates (increasing the minimum version).

**Comment by dfunckt** [comment]
Yes, resinOS containing Supervisors >=6.2.0 and <6.3.3 is better to be avoided, especially if the user's image is known to be large. I think there's no resinOS release containing these Supervisor versions in production though, but I haven't double checked.

**Comment by dfunckt** [comment]
@imrehg -- Just checked and we don't have resinOS releases containing affected Supervisors:

<img width="335" alt="screen shot 2017-10-17 at 10 47 21" src="https://user-images.githubusercontent.com/130359/31652878-55f323dc-b329-11e7-97d7-f9655ab5782f.png">

Might be good however to withdraw the affected Supervisor versions.

**Comment by imrehg** [comment]
@dfunckt we do have affected version in BeagleBone: [2.4.2+rev1](https://github.com/resin-os/meta-resin/blob/master/CHANGELOG.md#v242) came with 6.2.5 by default. What I can do, that if it's a beaglebone update and the target is that version, then upgrade to 6.3.5 instead (another polyfill for resinhup:)

**Comment by imrehg** [comment]
resinOS 2.7.3 will include the fixed 6.3.5 supervisor, this is probably good to close...
